src/Voron/Data/BTrees/TreeCursor.cs

```diff
@@ -46,78 +24,45 @@ public void Dispose()
         // The bulk of the clean-up code is implemented in Dispose(bool)
         private void Dispose(bool disposing)
         {
-            if (disposing)
-            {
-                _state.Clear();
-                _pagesByNumPool.Free(_state);
-            }
+            if (disposing == false) 
+                return;
+
+            _statePages.WeakClear();
+            _treePageStackPool.Free(_statePages);
         }
 
         public void Update(FastStack<TreePage> stack, TreePage newVal)
         {
-            var oldNode = stack.Pop();
-            stack.Push(newVal);
-
-            var oldPageNumber = oldNode.PageNumber;
-            var newPageNumber = newVal.PageNumber;
-
-            if (oldPageNumber == newPageNumber)
-            {
-                _state.PageByNum[oldPageNumber] = newVal;
-                return;
-            }
-
-            _anyOverrides = true;
-            _state.PageByNum[oldPageNumber] = newVal;
```

Comment:
So we didn't have to keep `PageByNum` at all?

Reply:
Nope. Not being used anywhere.